keys/spans.go

```diff
@@ -25,6 +29,12 @@ type Span struct {
 	Start, End proto.Key
 }
 
+// ContainsKey returns true if the span contains 'key'.
+func (s *Span) ContainsKey(key proto.Key) bool {
+	addr := KeyAddress(key)
+	return bytes.Compare(addr, s.Start) >= 0 && bytes.Compare(addr, s.End) < 0
```

Comment:
You should really be using the Key type's comparison functions instead of bytes.Compare.